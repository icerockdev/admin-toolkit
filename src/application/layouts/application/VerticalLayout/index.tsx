/* Copyright (c) 2020-2021 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license. */

import React, { FC } from 'react';
import { observer } from 'mobx-react';
import classNames from 'classnames';
import styles from './styles.module.scss';
import { useConfig } from '~/application/utils/hooks';
import { Link } from 'react-router-dom';
import { VerticalNavigation } from '~/containers/layout/VerticalNavigation';
import { Account } from '~/containers/auth/Account';

interface IProps {}

const VerticalLayout: FC<IProps> = observer(({ children }) => {
  const config = useConfig();

  return (
    <div className={classNames(styles.layout, 'vertical-layout')}>
      <div className={classNames(styles.menu, 'vertical-layout__menu')}>
        <div className={classNames(styles.logo, 'vertical-layout__logo')}>
          <Link to={config.fallbackUrl}>
            <img src={config.logo}  alt="logo"/>
          </Link>
        </div>

        <div
          className={classNames(
            styles.navigation,
            'vertical-layout__navigation'
          )}
        >
          <VerticalNavigation />
        </div>

        {!!config.auth && (
          <div
            className={classNames(styles.account, 'vertical-layout__account')}
          >
            {!!config.auth && <Account />}
          </div>
        )}
      </div>

      <div className={classNames(styles.content, 'vertical-layout__content')}>
        {children}
      </div>
    </div>
  );
});

export { VerticalLayout };
