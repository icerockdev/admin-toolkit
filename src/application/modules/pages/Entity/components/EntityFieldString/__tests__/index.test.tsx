/* Copyright (c) 2021 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license. */

import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { EntityFieldString } from "~/application";
import { observable } from "mobx";

test('EntityFieldString view test', () => {
  const element = <EntityFieldString
    name="Test string field"
    value="Text with https://icerock.dev link"
    fields={[]}
  />

  const component = renderer.create(element);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('EntityFieldString edit test', () => {
  const store = new class {
    @observable value = 'Some text';
  }();

  const handler = (value) => store.value = value

  const element = <EntityFieldString
    name="Test string field"
    value={store.value}
    isEditing={true}
    handler={handler}
    fields={[]}
  />

  const component = renderer.create(element);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  const wrapper = mount(element);

  expect(store.value).toEqual('Some text');

  wrapper.find('input').simulate('focus');
  wrapper.find('input').simulate('change',{target: {value: 'Some text changed'}});

  expect(store.value).toEqual('Some text changed');
});
