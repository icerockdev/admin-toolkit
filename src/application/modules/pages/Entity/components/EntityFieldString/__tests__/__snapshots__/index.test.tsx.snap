// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityFieldString edit test 1`] = `
<div>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=""
        required={false}
        type="text"
        value="Some text"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-2"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`EntityFieldString view test 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "Text with <a href=\\"https://icerock.dev\\" target=\\"blank\\" rel=\\"nofollow\\">https://icerock.dev</a> link",
    }
  }
/>
`;
